ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java

```diff
@@ -112,6 +113,8 @@
   public static final String ROUTER_REQUEST_HANDLER_NUM_OF_THREADS = "router.request.handler.num.of.threads";
   public static final String ROUTER_STORE_KEY_CONVERTER_FACTORY = "router.store.key.converter.factory";
   public static final String ROUTER_UNAVAILABLE_DUE_TO_OFFLINE_REPLICAS = "router.unavailable.due.to.offline.replicas";
+  public static final String ROUTER_CACHE_TTL_FOR_NOT_FOUND_RESPONSES_MS =
+      "router.cache.ttl.for.not.found.responses.ms";
```

Comment:
I don't quite like the configuration name, it has nothing to do with response, can we rename it to router.not.found.cache.ttl.in.ms", something like that?